apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -199,6 +204,33 @@ fn validate_source(directive: &Component<Directive>, sources: &SourceMap) -> Sou
     if let Some(url_error) = url_error {
         errors.push(url_error);
     }
+
+    if let Some(http_arg) = directive
```

Comment:
We already get the http arg up above, since we need it to validate the URL. So maybe we should get it once, earlier, and return early (probably with a `GraphQLError`, since the arg is required) if it's not present.